torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -427,6 +436,126 @@ TensorView* TensorView::computeAt(
   return this;
 }
 
+void TensorView::computeWith(
+    TensorView* consumer,
+    int consumer_pos,
+    bool best_effort) {
+  TORCH_INTERNAL_ASSERT(
+      !container()->isA<kir::Kernel>(),
+      "Function invalid for kernel container.");
+
+  if (isFusionInput()) {
+    return;
+  }
+
+  auto consumers = ir_utils::consumerTvsOf(this);
+  TORCH_CHECK(
+      std::find(consumers.begin(), consumers.end(), consumer) !=
+          consumers.end(),
+      "Not a direct consumer: ",
+      consumer->toString());
+
+  if (consumer_pos < 0) {
+    consumer_pos += int(consumer->nDims()) + 1;
+  }
+
+  TORCH_INTERNAL_ASSERT(
+      consumer_pos >= 0 && consumer_pos <= nDims(),
+      "Invalid inline position for ",
+      consumer->toString(),
+      ": ",
+      consumer_pos);
+
+  // Find the corresponding position in this tensor
```

Comment:
In terms of interface, which do you think makes more sense? `computeAt` uses consumer position, but `inlineAt` uses this position.

Reply:
`computeAt` as well as `computeWith` have a consumer parameter, and I thought it'd be more intuitive to have the position parameter relative to the consumer.

`inlineAt`, on the other hand, doesn't reference any particular consumer, and it makes sense that the position references its domain.

That said, the difference does seem confusing. If we wanted to have a uniform position semantic, I'd change `computeWith` to take a position in its domain.

Reply:
I have no idea on which will be more convenient to use in the scheduler, and I have no strong opinion about which should be used here.

Reply:
Most likely, the best-effort option should just work, so I think this difference won't matter much. I'm leaning toward changing this to the position in its own tensor as that's the position we ultimately save in each tensor (i.e., `compute_with_pos_`)